app/wrappers.ts

```diff
@@ -0,0 +1,178 @@
+  icons: Record<'16' | '32' | '48' | '128' | string, string>;
+  options_ui?: {
+    page: string;
+  };
+  options_page?: string;
+  homepage_url: string;
+  manifest_version: number;
+  key: string;
+  minimum_chrome_version: string;
+  action: unknown;
+  content_scripts: unknown[];
+  background: unknown;
+  web_accessible_resources: unknown[];
+  permissions: string[];
+  host_permissions: string[];
+  externally_connectable: unknown;
+  content_security_policy: unknown;
+}
+
+interface WalletManifest extends ExtensionManifest {
+  origin: string;
+}
+
+// A proposal function for @penumbra-zone/client
+export const getWallets = () => {
+  const injection = assertGlobalPresent();
+  return Object.keys(injection);
+};
+
+
+// A proposal function for @penumbra-zone/react
+export const useWallets = () => {
+  const [wallets, setWallets] = useState<string[]>([]);
+
+  useEffect(() => {
+    setWallets(getWallets());
+  }, []);
+
+  return wallets;
+};
+
+// Right now, we can't fetch manifests from @penumbra-zone/client, especially with types
+const fetchManifest = async (wallet: string): Promise<WalletManifest | undefined> => {
+  try {
+    const provider = assertProvider(wallet);
+    const req = await fetch(provider.manifest);
+    const manifest: ExtensionManifest = await req.json();
+    return { ...manifest, origin: wallet };
+  } catch (_) {
+    return undefined;
+  }
+};
+
+// Common react api for fetching wallet data to render the list of injected wallets
+export const useWalletManifests = () => {
+  const [manifests, setManifests] = useState<WalletManifest[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const fetchManifests = async () => {
+    setLoading(true);
+    const wallets = getWallets();
+    const res = await Promise.all(wallets.map(fetchManifest));
+    const manifests = res.filter(Boolean) as WalletManifest[];
+    setManifests(manifests);
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    fetchManifests();
+  }, []);
+
+  return { data: manifests, loading };
+};
+
+export const connectToWallet = (origin: string) => {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const provider = await assertProviderManifest(origin);
+
+      // Should use watchers and callbacks from the injection
+      const interval = setInterval(() => {
+        if (provider.isConnected()) {
+          clearInterval(interval);
+          resolve(true);
+        }
+      }, 10);
+
+      setTimeout(() => {
+        clearInterval(interval);
+        reject(new Error('could not connect to the wallet'));
+      }, 15000);
+
+      provider.request();
+    } catch (error) {
+      reject(error);
+    }
+  });
+};
+
+export const useConnect = () => {
+  const [connectionLoading, setConnectionLoading] = useState<boolean>(false);
+  const [connected, setConnected] = useState<string>();
+
+  const [address, setAddress] = useState<string>();
+  const [balances, setBalances] = useState<string[]>([]);
```

Comment:
These feel like they belong in another hook/component, separate than the connection stuff